server/utils/utils.ts

```diff
@@ -48,3 +48,18 @@ export function assertHasStringValues<K extends keyof Record<string, unknown>>(
     throw Error(`Missing or invalid keys: ${invalidKeys}`)
   }
 }
+
+export function assertHasOptionalStringValues<K extends string>(
+  obj: unknown,
+  keysToCheck: K[]
+): asserts obj is Record<K, string | undefined> {
```

Comment:
version of `assertHasStringValues` which copes with fields being absent. 

this is a type assertion to allow safe access of fields on unknown types:
```
const a: unknown = ...
assertHasOptionalStringValues(a, ['name'])
console.log(a.name)   // usually you wouldn't be able to access this
```